lib/new_relic/agent/instrumentation/grpc/server/instrumentation.rb

```diff
@@ -12,19 +12,23 @@ module GRPC
         module Server
           include NewRelic::Agent::Instrumentation::GRPC::Helper
 
+          DT_KEYS = [NewRelic::NEWRELIC_KEY, NewRelic::TRACEPARENT_KEY, NewRelic::TRACESTATE_KEY].freeze
           INSTANCE_VAR_HOST = :@host_nr
           INSTANCE_VAR_PORT = :@port_nr
           INSTANCE_VAR_METHOD = :@method_nr
+          CATEGORY = :web
+          DESTINATIONS = AttributeFilter::DST_TRANSACTION_TRACER |
+            AttributeFilter::DST_TRANSACTION_EVENTS |
+            AttributeFilter::DST_ERROR_COLLECTOR
 
-          def handle_with_tracing(active_call, mth, _inter_ctx)
+          def handle_with_tracing(streamer_type, active_call, mth, _inter_ctx)
             return yield unless trace_with_newrelic?
-            metadata = metadata_for_call(active_call)
 
-            finishable = NewRelic::Agent::Tracer.start_transaction_or_segment(
-              name: mth.original_name,
-              category: :web,
-              options: server_options(metadata)
-            )
```

Comment:
- For gRPC servers, go lower level when creating a transaction and use
  `start_new_transaction` instead of `start_transaction_or_segment`.